client/src/Pages/Food/index.js

```diff
@@ -1,10 +1,101 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import IconLabeledButton from '../../components/Button';
+import ContainerComponent from '../../components/Container';
+import TextInputField from '../../components/InputFeild';
+import CardComponent from '../../components/Card';
+
+const axios = require('axios');
 
 function FoodPage() {
+  const [allFood, setAllFood] = useState({});
+  const [errorMessage, setErrorMessage] = useState('');
+  const [searchValue, setSearchValue] = useState('');
+  const [foodFilter, setFoodFilter] = useState('');
+  useEffect(() => {
+    const { CancelToken } = axios;
+    const source = CancelToken.source();
+    (async () => {
+      try {
+        const food = await axios.get('/api/v1/food');
+        return setAllFood(food);
+      } catch (err) {
+        return setErrorMessage(
+          err.response.data.message || 'something went wrong'
+        );
+      }
+    })();
+    return () => source.cancel('Operation canceled by the user.');
+  });
   return (
-    <div>
+    <>
       <h1>Food Page</h1>
-    </div>
+      {errorMessage ? (
+        <p>{errorMessage}</p>
+      ) : (
+        <>
+          <TextInputField
+            type="text"
+            className="searchInput"
+            label="Search"
+            variant="filled"
+            onChange={(e) => setSearchValue(e.target.value)}
+            placeholder="search for your desired food"
+          />
+          <ContainerComponent
+            direction="row"
+            spacing={9}
+            screenSize="sm"
+            itemColumns="2"
+          >
+            <IconLabeledButton
+              variant="outlined"
+              disable={false}
+              className="filterFoodButton"
+              event={setFoodFilter('meal')}
+            >
+              Meals
+            </IconLabeledButton>
+            <IconLabeledButton
+              variant="outlined"
+              disable={false}
+              className="filterFoodButton"
+              event={setFoodFilter('food')}
+            >
+              Food
+            </IconLabeledButton>
+          </ContainerComponent>
+          <ContainerComponent
+            direction="column"
+            spacing={9}
+            screenSize="sm"
+            itemColumns="2"
+          >
+            <CardComponent
+              cardClassName="foodCard"
+              ContentClassName="innerFoodContent"
+            >
+              {allFood.data !== 0
+                ? allFood.data
+                    .map((row) => (
+                      <>
+                        <p>{row.food_name}</p>
+                        <p>{row.food_type_id}</p>
+                        <img src={row.image} alt={row.food_name} />
+                      </>
+                    ))
+                    .filter(
```

Comment:
I think usually the pattern is to first `filter` and then `map`.  It can be more efficient that way, since you are only mapping the items you need, instead of mapping every item and then throwing some away in the filter operation.

Reply:
I have two filtring operations, one for the name of the food and one for the type of the food and I need to apply these two filters to what foods will be viewed, I would appreciate any suggestions on how to do it.
my first thought was to perform 2 filters separately, one for the food type then, for the name.

Reply:
That would work yes. A slightly more efficient way would be to check both things (type and name) inside one filter call. Let me know if that works.